src/Runner.Worker/ActionManager.cs

```diff
@@ -101,38 +101,41 @@ public async Task<PrepareResult> PrepareActionsAsync(IExecutionContext execution
             IEnumerable<Pipelines.ActionStep> actions = steps.OfType<Pipelines.ActionStep>();
             executionContext.Output("Prepare all required actions");
             var result = await PrepareActionsRecursiveAsync(executionContext, state, actions, depth, rootStepId);
-            if (state.ImagesToPull.Count > 0)
+            if (FeatureFlagManager.IsHookFeatureEnabled() && FeatureFlagManager.IsCustomHookFeatureEnabled())
```

Comment:
I'm a little behind on this feature - the point is, that we prebuild step containers in the 'normal' runner, but in the 'hook' runner we want to pull / build them JIT (what's more, as part of the hook.[js]), right? 

If I got that right, shouldn't the if say `if(!hooksRunner)` then prebuild?

Reply:
My mistake, it's corrected now